desktop/libs/notebook/src/notebook/connectors/spark_shell.py

```diff
@@ -274,29 +321,121 @@ def get_jobs(self, notebook, snippet, logs):
+    session_key = self._get_session_key()
+    
+    session = SESSIONS[session_key] if SESSIONS.get(session_key) else self.create_session(snippet.get('type'))
+
+    if database is None:
+      response['databases'] = self._show_databases(api, session)
+    elif table is None:
+      response['tables_meta'] = self._show_tables(api, session, database)
+    elif column is None:
+      columns = self._get_columns(api, session, database, table)
+      response['columns'] = [col['name'] for col in columns]
+      response['extended_columns'] = [{
+          'comment': col.get('comment'),
+          'name': col.get('name'),
+          'type': col['type']
+        }
+        for col in columns
+      ]
 
     return response
 
-  def get_sample_data(self, snippet, database=None, table=None, column=None, is_async=False, operation=None):
-    if operation != 'hello':
-      raise NotImplementedError()
 
-    response = {}
+  def _check_status_and_fetch_result(self, api, session, execute_resp):
+    check_status = api.fetch_data(session['id'], execute_resp['id'])
+
+    while check_status['state'] in ['running', 'waiting']:
+      check_status = api.fetch_data(session['id'], execute_resp['id'])
+      time.sleep(1)
+
+    if check_status['state'] == 'available':
+      return self._fetch_result(api, session, execute_resp['id'], start_over=True)
+
+
+  def _show_databases(self, api, session):
+    show_db_execute = self._execute(api, session, 'SHOW DATABASES')
+    db_list = self._check_status_and_fetch_result(api, session, show_db_execute)
+
+    if db_list:
+      return [db[0] for db in db_list['data']]
+
+
+  def _show_tables(self, api, session, database):
+    use_db_execute = self._execute(api, session, 'USE %(database)s' % {'database': database})
+    use_db_resp = self._check_status_and_fetch_result(api, session, use_db_execute)
+
+    show_tables_execute = self._execute(api, session, 'SHOW TABLES')
+    tables_list = self._check_status_and_fetch_result(api, session, show_tables_execute)
+
+    if tables_list:
+      return [table[1] for table in tables_list['data']]
+
 
+  def _get_columns(self, api, session, database, table):
+    use_db_execute = self._execute(api, session, 'USE %(database)s' % {'database': database})
+    use_db_resp = self._check_status_and_fetch_result(api, session, use_db_execute)
+
+    describe_tables_execute = self._execute(api, session, 'DESCRIBE %(table)s' % {'table': table})
+    columns_list = self._check_status_and_fetch_result(api, session, describe_tables_execute)
+
+    if columns_list:
+      return [{
+        'name': col[0],
+        'type': col[1],
+        'comment': '',
+      } for col in columns_list['data']]
+
+
+  def get_sample_data(self, snippet, database=None, table=None, column=None, is_async=False, operation=None):
     api = self.get_api()
+    session_key = self._get_session_key()
+    session = SESSIONS.get(session_key)
+
+    if operation == 'hello':
+      statement = "SELECT 'Hello World!'"
```

Comment:
is it possible to include this condition in _get_select_query function

Reply:
Done